nucleus/payara-modules/hazelcast-bootstrap/src/main/java/fish/payara/nucleus/hazelcast/MemberAddressPicker.java

```diff
@@ -0,0 +1,158 @@
+ * file and include the License file at glassfish/legal/LICENSE.txt.
+ *
+ * GPL Classpath Exception:
+ * The Payara Foundation designates this particular file as subject to the "Classpath"
+ * exception as provided by the Payara Foundation in the GPL Version 2 section of the License
+ * file that accompanied this code.
+ *
+ * Modifications:
+ * If applicable, add the following below the License Header, with the fields
+ * enclosed by brackets [] replaced by your own identifying information:
+ * "Portions Copyright [year] [name of copyright owner]"
+ *
+ * Contributor(s):
+ * If you wish your version of this file to be governed by only the CDDL or
+ * only the GPL Version 2, indicate your decision by adding "[Contributor]
+ * elects to include this software in this distribution under the [CDDL or GPL
+ * Version 2] license."  If you don't indicate a single choice of license, a
+ * recipient has the option to distribute your version of this file under
+ * either the CDDL, the GPL Version 2 or to extend the choice of license to
+ * its licensees as provided above.  However, if you add GPL Version 2 code
+ * and therefore, elected the GPL Version 2 license, then the option applies
+ * only if the new code is made subject to such option by the copyright
+ * holder.
+ */
+package fish.payara.nucleus.hazelcast;
+
+import com.hazelcast.spi.MemberAddressProvider;
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.util.Enumeration;
+import java.util.HashSet;
+import java.util.logging.Level;
+import org.glassfish.api.admin.ServerEnvironment;
+import java.util.logging.Logger;
+
+/**
+ * This class tries to work out which interface to choose as the address of the member
+ * broadcast within Hazelcast
+ * @author Steve Millidge (Payara Foundation)
+ */
+public class MemberAddressPicker implements MemberAddressProvider {
+
+    private final ServerEnvironment env;
+    private final HazelcastRuntimeConfiguration config;
+    private final HazelcastConfigSpecificConfiguration localConfig;
+    private InetSocketAddress bindAddress;
+    private InetSocketAddress publicAddress;
+    private static final Logger logger = Logger.getLogger(MemberAddressPicker.class.getName());
+    
+    MemberAddressPicker(ServerEnvironment env, HazelcastRuntimeConfiguration config, HazelcastConfigSpecificConfiguration localConfig) {
+        this.env = env;
+        this.config = config;
+        this.localConfig = localConfig;
+        
+        // determine public address and bind address
+        findAppropriateInterfaces();
+    }
+
+    @Override
+    public InetSocketAddress getBindAddress() {
+        return bindAddress;
+    }
+
+    @Override
+    public InetSocketAddress getPublicAddress() {
+        if (publicAddress != null) {
+            return publicAddress;
+        } else {
```

Comment:
It's a matter of style, but the else is not needed here. Can return directly after the if block.